metapkg/tools/git.py

```diff
@@ -63,6 +65,18 @@ def update_repo(
     if ref == "HEAD":
         ref = None
 
+    if not clean_checkout:
+        checkout = (
+            GitBackend.get_default_source_root()
+            / GitBackend.get_name_from_source_url(repo_url)
+        )
+
+        if checkout.exists() and GitBackend.get_remote_url(
+            dulwich_repo.Repo(str(checkout))
+        ):
```

Comment:
I don't follow how this detects the URL having changed? it's not comparing anything...it'll be true anytime `get_remote_url` returns a true-ish value?

Reply:
:man_facepalming: